waspc/data/Generator/templates/server/src/auth/providers/oauth/createRouter.ts

```diff
@@ -43,47 +46,43 @@ export function createRouter(provider: ProviderConfig, initData: { passportStrat
           }
 
           // Wrap call to getUserFieldsFn so we can invoke only if needed.
-          const getUserFields = () => getUserFieldsFn(contextWithUserEntity, { profile: providerProfile });
+          const getUserFields = () => getUserFieldsFn ? getUserFieldsFn(contextWithUserEntity, { profile: providerProfile }) : Promise.resolve({});
           // TODO: In the future we could make this configurable, possibly associating an external account
           // with the currently logged in account, or by some DB lookup.
-          const user = await findOrCreateUserByExternalAuthAssociation(provider.id, providerProfile.id, getUserFields);
+          const auth = await findOrCreateAuthByAuthIdentity(provider.id, providerProfile.id, getUserFields);
 
-          const token = await sign(user.id);
+          const token = await sign(auth.{= userFieldOnAuthEntityName =}.id);
```

Comment:
userFieldOnAuthEntityName also sounds like something we might be ok with just hardcoding hm.

Reply:
I wanted to keep the source of truth in Haskell 🤷

Reply:
As I said above, it makes sense, but I don't think I would do it in this case. Unless  you see some reasons that I don't.

Reply:
Ok I see the reasons now that I read the whole PR.